slave-base/contrib/bin/run-jnlp-client

```diff
@@ -34,6 +34,28 @@ fi
 
 echo "OPENSHIFT_JENKINS_JVM_ARCH='${OPENSHIFT_JENKINS_JVM_ARCH}', CONTAINER_MEMORY_IN_MB='${CONTAINER_MEMORY_IN_MB}', using $(readlink /etc/alternatives/java) and $(readlink /etc/alternatives/javac)"
 
+if [[ -z "${NOOP_NO_PROXY_DEFAULT}" ]]; then
+    # we do not want jenkins svc or jenkins-jnlp svc
+    # communication going through a http proxy
+    # env vars to consider:
+    # - no_proxy and NO_PROXY; case of string varies tool to tool
+    # - JENKINS_URL and JENKINS_TUNNEL comes from k8s plugin
+    # based on how our master image configures the cloud, but we need to strip the host / port
+    jenkins_http_host=`echo $JENKINS_URL | sed 's/https:\/\///' | sed 's/http:\/\///' | cut -f1 -d":"`
+    jnlp_http_host=`echo $JENKINS_TUNNEL | sed 's/https:\/\///' | sed 's/http:\/\///' | cut -f1 -d":"`
```

Comment:
little tip when using sed:  you can pick the separator character so you don't have to escape your backslashed, for better readability, for example:
```
$ echo http://foobar.com | sed 's#http://#stripped#'
strippedfoobar.com
```